src/Sylius/Bundle/AdminBundle/spec/EventListener/AdminFilterSubscriberSpec.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\AdminBundle\EventListener;
+
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Sylius\Bundle\AdminBundle\Storage\FilterStorageInterface;
+use Symfony\Component\HttpFoundation\InputBag;
+use Symfony\Component\HttpFoundation\ParameterBag;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+final class AdminFilterSubscriberSpec extends ObjectBehavior
+{
+    function let(FilterStorageInterface $filterStorage): void
+    {
+        $this->beConstructedWith($filterStorage);
+    }
+
+    function it_subscribes_to_kernel_request_event(): void
+    {
+        $this::getSubscribedEvents()->shouldReturn([KernelEvents::REQUEST => 'onKernelRequest']);
+    }
+
+    function it_adds_filter_to_filter_storage(
+        RequestEvent $event,
+        Request $request,
+        ParameterBag $attributes,
+        ParameterBag $query,
+        FilterStorageInterface $filterStorage
```

Comment:
```suggestion
        FilterStorageInterface $filterStorage,
```
same with the rest functions here